pennylane/ops/op_math/adjoint.py

```diff
@@ -217,45 +213,23 @@ def _unflatten(cls, data, _):
 
     # pylint: disable=unused-argument
     def __new__(cls, base=None, id=None):
```

Comment:
is this new `__new__` breaking, in that `type(qml.adjoint(qml.PauliX(0))` is now `AdjointOpObs`? it's still a subclass of `Adjoint`, but it's different from before.

Reply:
Would this different type name be problematic anywhere down the pipeline? 🤔

Reply:
So before, `type(qml.adjoint(qml.PauliX(0))` would have been `Adjoint._operation_observable_type`, which also had the name `Adjoint`. So this is an internal detail that no one should be depending on. We didn't even really test it. We just made sure the instance was inherited from all the correct things.

Reply:
I am very fine with this remaining an internal detail and we do what's best here, but I do want everyone to agree on what's best and ensure that this PR and #4452 agree before the release. I think what you've done with `Adjoint` here is simpler, so it has my vote

Reply:
I think it was creating types with `type` that was causing problems.

Reply:
Debugging #4452, we found that the serialization for `Adjoint` as is now returns the wrong types for the child classes. We've fixed it in the `Projector` by overriding the `__new__` in the child classes for the standard python one:
```python
def __new__(cls):
    return object.__new__(cls)
```
Hopefully this works well 😄

Reply:
Thanks for tracking that down. I'll add that change.

Reply:
Any unit test I should do to make sure this accomplishes what it needs to?

Reply:
I have added a serialization test to my PR where I explicitly test that the final type matches the original one. Also, I was able to get rid of the custom `__copy__` method, since the subclasses no longer have a hacked `__new__`. I am not sure this'll be possible with `Adjoint` though